service/controller/legacy/v29/ebs/ebs.go

```diff
@@ -128,6 +128,43 @@ func (e *EBS) DetachVolume(ctx context.Context, volumeID string, attachment Volu
 		if err != nil {
 			return microerror.Mask(err)
 		}
+	}
+
+	{
+		i := &ec2.DescribeVolumesInput{
+			VolumeIds: []*string{
+				aws.String(volumeID),
+			},
+		}
+
+		v, err := e.client.DescribeVolumes(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if len(v.Volumes) == 0 {
+			// Fall through.
+			e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("volume %#q not found", volumeID))
+			return microerror.Maskf(executionFailedError, "no volume found")
+		}
+
+		d := v.Volumes[0]
+		if len(d.Attachments) > 1 {
+			// Shouldn't happen; log so we know if it is
```

Comment:
This is an approach borrowed from the upstream implementation. If the volume has multiple attachments does not mean that there is an error. More that we need to be aware of this and log. But at the end since we have a range below we'll loop through it